### PR TITLE
Add support for ingress gateway CRD

### PR DESCRIPTION
**Pull request by lkysow**
Companion to https://github.com/hashicorp/consul-k8s/pull/407

Also updates controller-tools to 0.4.1 and fixes typo in `serviceresolver` crd.

### CODE REVIEW

File: templates/crd-serviceresolvers.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ spec:
                     items:
                       type: string
                     type: array
-                  namespaces:
+                  namespace:
```
Comment (lkysow) [inline templates/crd-serviceresolvers.yaml:65]: todo: look at what effect his would have

File: .circleci/config.yml [hunk 2]
```diff
@@ -152,7 +152,8 @@ jobs:
                     -kubeconfig="$primary_kubeconfig" \
                     -secondary-kubeconfig="$secondary_kubeconfig" \
                     -debug-directory="$TEST_RESULTS/debug" \
-                    -consul-k8s-image=docker.mirror.hashicorp.services/hashicorpdev/consul-k8s:latest
+                    -consul-k8s-image=ghcr.io/lkysow/consul-k8s-dev:nov26 \
```
Comment (ndhanushkodi) [inline .circleci/config.yml:155]: Friendly reminder to remove when merging :)

File: templates/crd-serviceresolvers.yaml (hunk 1 above)
Comment (ndhanushkodi) [reply templates/crd-serviceresolvers.yaml:65]: Nice catch and thanks for the detailed explanation of the impact!

**Review by ndhanushkodi — APPROVED** [review]
Looks great 🎉 🚀

**Review by ashwin-venkatesh — APPROVED** [review]
sorry i missed this last time around!! looks great

File: templates/crd-serviceresolvers.yaml (hunk 1 above)
Comment (lkysow) [reply templates/crd-serviceresolvers.yaml:65]: The new helm copier caught it!

**Pull request merged**